src/proposer/proposer.cpp

```diff
@@ -114,26 +183,7 @@ class ProposerImpl : public Proposer {
           }
           wallet_ext.GetProposerState().m_status = Status::IS_PROPOSING;
           wallet_ext.GetProposerState().m_number_of_search_attempts += 1;
-          const boost::optional<EligibleCoin> &winning_ticket = m_proposer_logic->TryPropose(coins);
-          if (!winning_ticket) {
-            LogPrint(BCLog::PROPOSING, "Not proposing this time (wallet=%s)\n", wallet_name);
-            continue;
-          }
-          const EligibleCoin &coin = winning_ticket.get();
-          LogPrint(BCLog::PROPOSING, "Proposing... (wallet=%s, coin=%s)\n",
-                   wallet_name, util::to_string(coin.utxo));
-          staking::TransactionPicker::PickTransactionsParameters parameters{};
-          staking::TransactionPicker::PickTransactionsResult result =
-              m_transaction_picker->PickTransactions(parameters);
-
-          if (!result) {
-            LogPrint(BCLog::PROPOSING, "Failed to pick transactions (wallet=%s, error=%s) – proposing empty block.\n", wallet_name, result.error);
-          }
-          const CAmount fees = std::accumulate(result.fees.begin(), result.fees.end(), CAmount(0));
-          const uint256 snapshot_hash = m_active_chain->ComputeSnapshotHash();
-
-          block = m_block_builder->BuildBlock(
-              tip, snapshot_hash, coin, coins, result.transactions, fees, wallet_ext);
+          GenerateBlock(wallet.get(), tip, coins, boost::none, block);
```

Comment:
Returned value is ignored.

Reply:
I'll get rid of the return value as is redundant with the out param.